common/signature/signature.go

```diff
@@ -24,13 +24,22 @@ import (
 	"fmt"
 )
 
-// Sign generates a base64-encoded signature of the input data using the provided RSA private key.
-func Sign(privateKey *rsa.PrivateKey, input []byte) (string, error) {
+// Sign generates a base64-encoded signature of the input data using the provided private key.
+// The private key must be RSA.
+func Sign(privateKey crypto.PrivateKey, input []byte) (string, error) {
 	hashed := sha256.Sum256(input)
-	sig, err := rsa.SignPKCS1v15(nil, privateKey, crypto.SHA256, hashed[:])
-	if err != nil {
-		return "", fmt.Errorf("Sign(): unable to sign signature: %w", err)
+	var sig []byte
+	var err error
+	switch priv := privateKey.(type) {
```

Comment:
omg it is CHAOTIC to me that switch statements don't need break statements 
![image](https://github.com/openconfig/bootz/assets/142375860/0ccaceff-4d95-445e-9985-a76e847e3611)

Reply:
Just Go things :shrug: